lib/backend/syncersv1/updateprocessors/rules.go

```diff
@@ -154,6 +203,47 @@ func parseNamespaceSelector(s string) string {
 	return updated
 }
 
+// parseServiceAccounts takes a v2 service account selector and returns the appropriate v1 representation
+// by prefixing the keys with the `pcsa/` prefix. For example, `k == 'v'` becomes `pcsa/k == 'v'`.
+func parseServiceAccounts(sam *apiv3.ServiceAccountMatch) string {
+
+	parsedSelector, err := parser.Parse(sam.Selector)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to parse service account selector: %s", sam.Names)
+		return ""
+	}
+
+	parsedSelector.AcceptVisitor(parser.PrefixVisitor{Prefix: conversion.ServiceAccountLabelPrefix})
+	updated := parsedSelector.String()
+	log.WithFields(log.Fields{"original": sam.Selector, "updated": updated}).Debug("Updated service account selector")
+	if len(sam.Names) == 0 {
+		return updated
+	}
+
+	// Convert the list of ServiceAccounts to selector
+	var namesSelector, comma string
+	namesSelector = fmt.Sprintf("%s in { ", apiv3.LabelServiceAccount)
+	for _, name := range sam.Names {
```

Comment:
Isn't Names a string?

Reply:
Ah I see, I was looking at the v1 definition - which I think is surplus code.

Since Names is just a strings slice, you can use strings.Join to join together the list of values with a ", " delimiter.

Reply:
phew... so to confirm i should nuke the code in v1/rule.go?! Right.
i'll use stings.Join.